lib/chef/audit/audit_reporter.rb

```diff
@@ -116,8 +118,14 @@ def post_auditing_data(error = nil)
         Chef::Log.debug("Sending audit report (run-id: #{audit_data.run_id})")
         run_data = audit_data.to_hash
 
-        if error
-          run_data[:error] = "#{error.class.to_s}: #{error.message}\n#{error.backtrace.join("\n")}"
+        if @exception || error
+          errors = [@exception, error].uniq.compact
+          errors_messages = errors.map  do |err|
+            msg = "#{err.class.to_s}: #{err.message}"
+            msg << "\n#{err.backtrace.join("\n")}" if err.backtrace
+            msg
+          end
```

Comment:
@tyler-ball reading my mind from over there :smile: 

We had a chat with @mcquin and that's exactly what we think we should do.